lua/ulx/modules/sh/cfc_curse.lua

```diff
@@ -9,6 +9,36 @@ do -- Load curse effects
 end
 
 
+local function getDurationStrings( durationSeconds, effectOverride )
+    local hasCustomDuration = durationSeconds > 0 and ( not effectOverride or not effectOverride.blockCustomDuration )
+    local durationAppend = ""
+    local briefly = " briefly"
+
+    if hasCustomDuration then
+        local durationStr = durationSeconds >= 60 and
+            ULib.secondsToStringTime( durationSeconds ) or
+            math.Round( durationSeconds ) .. " seconds"
+
+        durationAppend = " for " .. durationStr
+        briefly = ""
+    end
+
+    return durationAppend, briefly
+end
+
+local function getOneTimeEffects()
+    local out = {}
+
+    for _, effect in ipairs( CFCUlxCurse.Effects ) do
+        if not effect.excludeFromOnetime then
+            table.insert( out, effect )
+        end
+    end
+
+    return out
+end
```

Comment:
Basic helper function for getting one-time effects, i.e. effects able to be chosen out of ulx curse via random selection. Should probably be moved to `sh_utils`.